src/fklearn/preprocessing/schema.py

```diff
@@ -0,0 +1,134 @@
+import functools
+import inspect
+import pandas as pd
+import toolz
+
+from typing import Any, Callable, Dict, List, Optional, Union
+from fklearn.types import LearnerLogType, LearnerReturnType
+
+
+@toolz.curry
+def feature_duplicator(
+    df: pd.DataFrame,
+    columns_to_duplicate: Optional[List[str]] = None,
+    columns_mapping: Optional[Dict[str, str]] = None,
+    preffix: Optional[str] = None,
+    suffix: Optional[str] = None
+) -> LearnerReturnType:
+    """
+    Duplicates some columns in the dataframe
+
+    Parameters
+    ----------
+    df: pandas.DataFrame
+        A Pandas' DataFrame with columns_to_duplicate columns
+
+    columns_to_duplicate: list of str
+        List of columns names
+
+    columns_mapping: int (default None)
+        Mapping of source columns to destination columns
+
+    preffix: int (default None)
+        Preffix to add to columns to duplicate
+
+    suffix: int (default None)
+        Suffix to add to columns to duplicate
+
+    Returns
+    ----------
+    increased_dataset : pandas.DataFrame
+        A dataset with repeated columns
+    """
+
+    columns_final_mapping = (
+        columns_mapping
+        if columns_mapping is not None
+        else {
+            col: (preffix or '') + str(col) + (suffix or '')
+            for col in columns_to_duplicate
+        }
+        if columns_to_duplicate
+        else dict()
+    )
+
+    def p(new_df: pd.DataFrame) -> pd.DataFrame:
+        for src_col, dest_col in columns_final_mapping.items():
+            new_df.insert(len(new_df.columns), dest_col, df[src_col])
+        return new_df
+
+    p.__doc__ = feature_duplicator.__doc__
+
+    log: LearnerLogType = {
+        'feature_duplicator': {
+            'columns_to_duplicate': columns_to_duplicate,
+            'columns_mapping': columns_mapping,
+            'preffix': preffix,
+            'suffix': suffix,
+            'columns_final_mapping': columns_final_mapping,
+        }
+    }
+
+    return p, p(df), log
```

Comment:
maybe here we should copy the Data Frame `df` otherwise this method will mutate the original dataset